packages/react/src/components/PageHeader/PageHeader.tsx

```diff
@@ -489,14 +503,150 @@ const PageHeaderTabBar = React.forwardRef<
     },
     className
   );
+  const [openPopover, setOpenPopover] = useState(false);
+  const tagSize = tags[0]?.size || 'md';
+
+  const tagsWithIds = useMemo(() => {
+    return tags.map((tag, index) => ({
+      ...tag,
+      id: tag.id || `tag-${index}`,
```

Comment:
It's unlikely but possible that `tag-${index}` won't be unique on the page. Anywhere we define `id`-type props/strings like this we can use the internal `useId()` helper:

```
import { useId } from '../../internal/useId';

...

const instanceId = useId('PageHeaderTabBar');
```

```suggestion
      id: tag.id || `tag-${index}-${instanceId}`,
```